### ISSUE TITLE
Test Content long duration for Test 9.6

### ISSUE THREAD

**Issue by louaybassbouss** [cta-wave/Test-Content#39]
DPCAT Test [9.6 Long Duration Playback](https://docs.google.com/document/d/1a4yroEi76U88k71iisdRzwhzcTFD7KxHHSJo4thsczA/edit#heading=h.bl78lv4pl5ay) requires long duration content. Not sure about appropriate duration  @dsilhavy  @jpiesing  any recommendation? I can remember a discussion of 1h content but not sure.

**Comment by rbouqueau** [comment]
I see it's been assigned to me but what can I do about it? What would be the associated mezzanine content?

As a proposal we could ask gpac to generate a one hour long beepbop counter a/o mix some continuous audio with it?

**Comment by jpiesing** [comment]
@nicholas-fr is working on the mezzanine content for this.

**Comment by cta-source** [comment]
For the audio portion, I think we discussed looping the PRN file. This gives a mod 60s timing result, which should be OK for most purposes. If something is off by more than 59 seconds, it is unlikely to be off by exactly 60s (or 120s, etc.), so the audio tests will still flag an error.

**Comment by rbouqueau** [comment]
Just to make sure I understand correctly: it is expected that we loop the audio input to match the video input duration, correct?

**Comment by jpiesing** [comment]
> Just to make sure I understand correctly: it is expected that we loop the audio input to match the video input duration, correct?

@rbouqueau See @cta-source 's input in https://github.com/cta-wave/mezzanine/issues/46

**Comment by cta-source** [comment]
Answered this question in https://github.com/cta-wave/mezzanine/issues/46.

**Comment by jpiesing** [comment]
@nicholas-fr has just uploaded some mezzanine for long duration, it just needs encoding. Long duration is a lower priority though than chunked content as long duration is only a 'nice to have' for the 1st release.

**Comment by gitwjr** [comment]
Deferred to future work.

**Comment by rbouqueau** [comment]
Is https://dash.akamaized.net/WAVE/Mezzanine/releases/3/tos_LD1_1920x1080@30_7200.mp4 considered like the source content for this issue?

**Comment by rbouqueau** [comment]
Also, could we update the sparse matrix to indicate which parameters to use? (I guess any, so I would copy them from T1?)

**Comment by rbouqueau** [comment]
PS: I made a test stream, with id="tLD" (name was inferred by script as 't' + 'LD' mezzanine prefix for Long Duration) using the existing script: all seems good. Not uploaded.

@nicholas-fr we may need to add some verification for this one too.

**Comment by nicholas-fr** [comment]
@rbouqueau Yes, the LD mezzanine were created for the long duration tests.
The agreed long duration streams/formats created are:
https://dash-large-files.akamaized.net/WAVE/Mezzanine/releases/4/tos_LD1_1920x1080@30_7200.mp4
https://dash-large-files.akamaized.net/WAVE/Mezzanine/releases/4/croatia_LD1_1920x1080@25_7200.mp4

It was agreed that a stream with a fractional frame rate was not needed.

**Comment by rbouqueau** [comment]
@nicholas-fr Any reason for not generating https://dash-large-files.akamaized.net/WAVE/Mezzanine/releases/4/tos_LD1_1920x1080@29.97_7200.mp4 ? If you do so I can integrate long duration stream in the main script (it already works). Otherwise it would have to stay in a separate script, which is less convenient for maintenance.

**Comment by rbouqueau** [comment]
Today's call: @nicholas-fr  can we reduce the size of the content? 2 frame rates x (unzipped + zip) = 4 x 30GB = 120GB per run.

**Comment by rbouqueau** [comment]
Content is now online and referenced by database.json.

**Comment by rbouqueau** [comment]
What prevents closing?

**Comment by gitwjr** [comment]
Closed. Content is online and referenced.

**Comment by jpiesing** [comment]
I am reopening the issue as it seems the audio was left out of the encoding. The mezzanine issue mentioned above says that the looped PN audio was created but if I look at https://cta-wave.github.io/Test-Content/, the MPD is at (https://dash.akamaized.net/WAVE/vectors/cfhd_sets/12.5_25_50/tLD/2023-09-01/stream.mpd) and this only includes video.

If the above is correct then tLD needs to be re-encoded.

**Comment by rbouqueau** [comment]
Who would encode the audio (and how)?

The script [allows to make a generic AAC-LC encoding](https://github.com/cta-wave/Test-Content-Generation/blob/637678562a92a3f4c187e3ac08c21fcae212304f/run-all.py#L56). However we discarded this approach for all our streams.

That also means that not only LD has no audio, but also the SS, chunked, and splicing contents.

**Comment by jpiesing** [comment]
> Who would encode the audio (and how)?
> 

Good question. Resillion's encoding only covers audio for audio-only tests.

> The script [allows to make a generic AAC-LC encoding](https://github.com/cta-wave/Test-Content-Generation/blob/637678562a92a3f4c187e3ac08c21fcae212304f/run-all.py#L56). However we discarded this approach for all our streams.
> 

Can we add a row to the sparse matrix to turn this on/off and modify the script to act on that row? 

> That also means that not only LD has no audio, but also the SS, chunked, and splicing contents.

We have two separate series of tests corresponding to sections 8.x and 9.x of the DPCTF specification. 
- Section 8.x tests are either video or audio but never both.
- Section 9.x tests need both video and audio. Here's section 9.

![image](https://github.com/cta-wave/Test-Content/assets/8490792/0a2a9272-9d45-40b0-87ab-46e9778e6d82)

- long duration playback is DPCTF spec section 9.6 so needs both video and audio. 
- SS and chunked tests are section 8 tests.
- Section 9.4 is a splicing test that needs video and audio in both splice_main and splice_ad. That one is blocked due to lack of this. See https://github.com/cta-wave/dpctf-tests/issues/143

**Comment by rbouqueau** [comment]
> Can we add a row to the sparse matrix to turn this on/off and modify the script to act on that row?

Yes, we can add a line in the configuration for the generation (in https://github.com/cta-wave/Test-Content-Generation/tree/637678562a92a3f4c187e3ac08c21fcae212304f/profiles).

Thanks for the clarifications. In https://github.com/cta-wave/dpctf-tests/issues/143 it seems that Resilion provided the audio content, correct? If so how was it generated?

If you need these, I can add the audio for LD + splicing and regenerate them. This will be basic AAC-LC though. Can you confirm this is what we want to do? I can generate it very soon.

**Comment by rbouqueau** [comment]
Ping

**Comment by rcottingham** [comment]
Resillion will generate the Long Duration test vector. Propose to loop PN02 (1min) for 2 hours.
Updated sparse matrix is here: https://github.com/cta-wave/Test-Content/issues/58#issuecomment-2631136601

**Comment by nicholas-fr** [comment]
The long duration mezzanine uses looped PN01.wav (following https://github.com/cta-wave/mezzanine/issues/46).
I believe it would be best for test vector content to not deviate from the mezzanine, otherwise we may lose track of how it was created. 
@rcottingham Any reason why PN02 would be preferred over PN01?

**Comment by rcottingham** [comment]
Long duration audio has now been generated and validated by Resillion, and verified by Fraunhofer IIS. Will provide to @nicholas-fr for upload to Akamai then we can generate a LD test.

**Comment by rbouqueau** [comment]
#78 merged.

**Comment by rbouqueau** [comment]
Is there anything remaining on this issue before I close it?

**Comment by rcottingham** [comment]
No - can be closed.
